src/sentry/api/endpoints/project_rule_details.py

```diff
@@ -25,10 +25,14 @@ def get(self, request, project, rule_id):
             {method} {path}
 
         """
-        rule = Rule.objects.get(
-            project=project, id=rule_id, status__in=[RuleStatus.ACTIVE, RuleStatus.INACTIVE]
-        )
-        return Response(serialize(rule, request.user))
+
+        try:
+            rule = Rule.objects.get(
+                project=project, id=rule_id, status__in=[RuleStatus.ACTIVE, RuleStatus.INACTIVE]
+            )
```

Comment:
You also need to do this in `put` and `delete`. The best option here is probably to implement `convert_args` so that we can have the rule automatically created and passed to `get/put/delete`.

Take a look at https://github.com/getsentry/sentry/blob/dced446ed7d95a259704bffa91f0393ef8774785/src/sentry/api/bases/incident.py#L28 for an example.